Code/Tools/AssetProcessor/native/resourcecompiler/RCQueueSortModel.cpp

```diff
@@ -61,7 +61,14 @@ namespace AssetProcessor
                     if (jobDepedencyInternal.m_jobDependency.m_type == AssetBuilderSDK::JobDependencyType::Order || jobDepedencyInternal.m_jobDependency.m_type == AssetBuilderSDK::JobDependencyType::OrderOnce)
                     {
                         const AssetBuilderSDK::JobDependency& jobDependency = jobDepedencyInternal.m_jobDependency;
-                        QueueElementID elementId(jobDependency.m_sourceFile.m_sourceFileDependencyPath.c_str(), jobDependency.m_platformIdentifier.c_str(), jobDependency.m_jobKey.c_str());
+                        AZ_Assert(
```

Comment:
It comes from the builders and is run through some processing code which is responsible for converting whatever type of input is provided into the expected format here.

Reply:
So my question still stands. This seems like some input that will happen again and again for the customer. The AZ_Assert() will crash this application due to bad input; and it seems that this asset builder will continue getting the same input. 

How can this detection inform the customer about how to avoid/fix this?

Reply:
Sorry I wasn't clear.  The customer can't fix this because it's a coding error.  This code assumes the previous processing code did it's job correctly (converting the various inputs into an absolute path), so the assert is there to catch that in case something changes (or a unit test hasn't set things up correctly...).  So this wouldn't happen because of a data error, only because someone changes the code.